test-data/unit/check-typeguard.test

```diff
@@ -602,12 +602,31 @@ def func(names: Tuple[str, ...]):
 from typing_extensions import TypeGuard
 
 class Z:
-    def typeguard(self, *, x: object) -> TypeGuard[int]:  # E: TypeGuard functions must have a positional argument
+    def typeguard1(self, *, x: object) -> TypeGuard[int]:  # line 4
+        ...
+    
+    @staticmethod
+    def typeguard2(x: object) -> TypeGuard[int]:
+        ...
+    
+    @staticmethod  # line 11
+    def typeguard3(*, x: object) -> TypeGuard[int]:
         ...
 
-def bad_typeguard(*, x: object) -> TypeGuard[int]:  # E: TypeGuard functions must have a positional argument
+def bad_typeguard(*, x: object) -> TypeGuard[int]: # line 15
     ...
+
+# In Python 3.8 the line number associated with FunctionDef nodes changed
 [builtins fixtures/tuple.pyi]
+[builtins fixtures/classmethod.pyi]
```

Comment:
Is it correct to have both of these builtins fixtures here, or should we delete one of `[builtins fixtures/tuple.pyi]` and `[builtins fixtures/classmethod.pyi]`?

Reply:
IIRC `tuple.pyi` is necessary for the typeguards and `classmethod.pyi` allows for the `@staticmethod` bits.

Reply:
Maybe I misunderstood how the test fixtures worked — I assumed it was only possible for mypy to use a single `builtins` fixture in any given test. Unless the test infrastructure has some fancy way of merging the two fixtures together to create a composite stub for `builtins.pyi` for this test?

Reply:
Huh, I didn't get that impression. I've just done stuff like this and it's worked like I expected. Let me check the outputs without them...

Huh, it works without `[builtins fixtures/tuple.pyi]`. Maybe `classmethod.pyi` includes that or something. I'll remove that from my prior test case that used it too!

Reply:
Should we check for duplicate file names? https://github.com/python/mypy/blob/3f35ebbb531ee04aa4b130c6574433b0943400e3/mypy/test/data.py#L96-L101

Reply:
Yeah, that might be helpful!

Reply:
Done, see #14960!

Reply:
Brilliant — thank you!